src/graphql/candyMachine.ts

```diff
@@ -659,15 +677,25 @@ export const CandyMachineUploadMutation = mutationField('candyMachineUpload', {
         .finally(async () => {
           removeStorageDir();
         });
+
+      if (args.executeSync) {
+        logger.info('Waiting for runUploadV2UsingHiddenSettings');
+        await uploadPromise;
+      }
     } else {
-      runUploadV2(logger, processId, args)
+      const uploadPromise = runUploadV2(logger, processId, args)
         .catch((err) => {
           logger.error('Aborting runUploadV2 due to error');
           logger.error(err);
         })
         .finally(async () => {
           removeStorageDir();
         });
+
+      if (args.executeSync) {
```

Comment:
It will return early (thus preventing a timeout when being called in prod), but the process still runs to completion. Its not great IMO, but i'm just preserving the existing funcationality here while giving myself a flag for my "test" to run.